File: java/concepts/Arrays.java
package concepts;

import java.util.ArrayList;
import java.util.List;

public class Arrays {
    public static void demonstrate() {
        System.out.println("\n=== Arrays & ArrayLists ===");

        // An ArrayList is a resizable array, which is very flexible.
        List<String> fruitsList = new ArrayList<>();
        fruitsList.add("apple");
        fruitsList.add("banana");
        fruitsList.add("orange");
        fruitsList.add("grape");

        System.out.println("ArrayList: " + fruitsList);

        // Access the first element.
        System.out.println("First fruit: " + fruitsList.get(0));

        // Access the last element using the .size() method.
        System.out.println("Last fruit: " + fruitsList.get(fruitsList.size() - 1));
    }
}

File: cpp/concepts/arrays.cpp
#include <iostream>
#include <string>
#include <vector> // Required for std::vector
#include "concepts.hpp"

// Arrays & Vectors
void demonstrateArrays() {
    std::cout << "\n=== Arrays & Vectors ===" << std::endl;

    // A std::vector is the modern, flexible, and recommended way to handle dynamic arrays.
    std::vector<std::string> fruits_vec = {"apple", "banana", "orange", "grape"};

    std::cout << "Vector (dynamic): ";
    for (const auto& fruit : fruits_vec) {
        std::cout << fruit << " ";
    }
    std::cout << std::endl;

    // Access the first element.
    std::cout << "First fruit of vector: " << fruits_vec[0] << std::endl;

    // Access the last element using the .size() method.
    std::cout << "Last fruit of vector: " << fruits_vec[fruits_vec.size() - 1] << std::endl;
}
